x-pack/test/fleet_api_integration/apis/space_awareness/space_awareness_migration.ts

```diff
@@ -0,0 +1,140 @@
+  const esClient = getService('es');
+  const kibanaServer = getService('kibanaServer');
+
+  describe('space awareness migration', async function () {
+    skipIfNoDockerRegistry(providerContext);
+    const apiClient = new SpaceTestApiClient(supertest);
+
+    before(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.savedObjects.cleanStandardList({
+        space: TEST_SPACE_1,
+      });
+      await cleanFleetIndices(esClient);
+    });
+
+    after(async () => {
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.savedObjects.cleanStandardList({
+        space: TEST_SPACE_1,
+      });
+      await cleanFleetIndices(esClient);
+    });
+
+    setupTestSpaces(providerContext);
+
+    // Create agent policies it should create a enrollment key for every keys
+    before(async () => {
+      const [defaultSpacePolicy1, spaceTest1Policy1] = await Promise.all([
+        apiClient.createAgentPolicy(),
+        apiClient.createAgentPolicy(TEST_SPACE_1),
+        apiClient.createAgentPolicy(TEST_SPACE_1),
+      ]);
+
+      await apiClient.installPackage({
+        pkgName: 'nginx',
+        pkgVersion: '1.20.0',
+        force: true, // To avoid package verification
+      });
+
+      await apiClient.createPackagePolicy(undefined, {
+        policy_ids: [defaultSpacePolicy1.item.id],
+        name: `test-nginx-1-${Date.now()}`,
+        description: 'test',
+        package: {
+          name: 'nginx',
+          version: '1.20.0',
+        },
+        inputs: {},
+      });
+
+      await apiClient.createPackagePolicy(TEST_SPACE_1, {
+        policy_ids: [spaceTest1Policy1.item.id],
+        name: `test-nginx-2-${Date.now()}`,
+        description: 'test',
+        package: {
+          name: 'nginx',
+          version: '1.20.0',
+        },
+        inputs: {},
+      });
+    });
+
+    describe('without opt-in', () => {
+      it('agent policies should not be space aware', async () => {
+        const policiesDefaultSpaceIds = (await apiClient.getAgentPolicies()).items
+          .map(({ id }) => id)
+          .sort();
+
+        const policiesTestSpaceIds = (await apiClient.getAgentPolicies(TEST_SPACE_1)).items
+          .map(({ id }) => id)
+          .sort();
+
+        expect(policiesDefaultSpaceIds.length).to.eql(3);
+        expect(policiesDefaultSpaceIds).to.eql(policiesTestSpaceIds);
+      });
+
+      it('package policies should not be space aware', async () => {
+        const policiesDefaultSpaceIds = (await apiClient.getPackagePolicies()).items
+          .map(({ id }) => id)
+          .sort();
+
+        const policiesTestSpaceIds = (await apiClient.getPackagePolicies(TEST_SPACE_1)).items
+          .map(({ id }) => id)
+          .sort();
+
+        expect(policiesDefaultSpaceIds.length).to.eql(2);
+        expect(policiesDefaultSpaceIds).to.eql(policiesTestSpaceIds);
+      });
+    });
+
+    describe('with space awareness opt-in', () => {
```

Comment:
Can we add a test case ensuring our lock is working as expected? This might be tricky for an FTR test but should be possible with a `jest:integration_test`. See `src/core/server/integration_tests` for examples...

Reply:
I added a test that test the concurrency here https://github.com/elastic/kibana/blob/adbd050bbb0bb2a6b77a505b707768956fcb10dc/x-pack/plugins/fleet/server/integration_tests/enable_space_awareness.test.ts#L193